### PR TITLE
*: pre-transition ign refactor

### PR DESCRIPTION
**Pull request by kikisdeliveryservice**
**- What I did**
Now that we have more time, I am breaking up PR #583 . This PR is removing the changes from #583 that can go into the code-base now to make the future transition easier. 

This PR makes the following changes:
- update helpers.go NewIgnConfig() to use MaxVersion and sub igntypes for ignv2_2types as package alias
- for all non-test files, substitute igntypes for ignv2_2types as package alias (these changes are trivial but take up a lot of space in the PR, let's do it now!)
- in test files, make the above substitution, and also refactor to use NewIgnConfig() instead of generating empty config and hardcoding version.
- minor doc update to reflect above changes

After this lands, the future subtantive ignition change PR should be clearer and easier to test and evaluate.

**- How to verify it**
This PR makes no functional changes and everything should work exactly the same as before.

### CODE REVIEW

File: lib/resourceapply/machineconfig_test.go [hunk 1]
```diff
@@ -175,9 +178,9 @@ func TestApplyMachineConfig(t *testing.T) {
 		input: &mcfgv1.MachineConfig{
 			ObjectMeta: metav1.ObjectMeta{Name: "foo"},
 			Spec: mcfgv1.MachineConfigSpec{
-				Config: ignv2_2types.Config{
-					Passwd: ignv2_2types.Passwd{
-						Users: []ignv2_2types.PasswdUser{{
+				Config: igntypes.Config{
+					Passwd: igntypes.Passwd{
+						Users: []igntypes.PasswdUser{{
 							HomeDir: "/home/dummy",
```
Comment (LorbusChris) [inline lib/resourceapply/machineconfig_test.go]: nit: Use the `homeDir1` here as well?

Comment (kikisdeliveryservice) [reply lib/resourceapply/machineconfig_test.go]: this is n/a now

**Comment by kikisdeliveryservice** [comment]
seeing:
```Apr 09 23:39:44 ip-10-0-3-216 bootkube.sh[8203]: "secret-initial-kube-controller-manager-service-account-private-key.yaml": failed to create: Post https://localhost:6443/api/v1/namespaces/openshift-config/secrets: dial tcp [::1]:6443: connect: connection refused
Apr 09 23:39:44 ip-10-0-3-216 bootkube.sh[8203]: Skipped /v1, Resource=secrets as it already exists
```
and
```"name": "cloud-credential",
                    {
                        "lastTransitionTime": "2019-04-09T23:21:23Z",
                        "message": "4 of 4 credentials requests are failing to sync.",
                        "reason": "CredentialsFailing",
                        "status": "True",
                        "type": "Failing"
                    },
```

/retest

**Comment by kikisdeliveryservice** [comment]
Not going to retest right now PR is hitting BZ# 1698253

see artifacts/e2e-aws-op/pods.json:
` "message": "Failed to inspect image \"registry.svc.ci.openshift.org/ci-op-0d9qlb6n/stable@sha256:8eb140d803ec324f5b3b472be7ffc2a6c582923833280c9d78227ccfb65d2154\": rpc error: code = Unknown desc = Manifest does not match provided manifest digest sha256:8eb140d803ec324f5b3b472be7ffc2a6c582923833280c9d78227ccfb65d2154"`,

**Comment by kikisdeliveryservice** [comment]
aws vpc issues, this is a known issue, going to hold off on retesting.

**Comment by kikisdeliveryservice** [comment]
ok trying again:

/test e2e-aws

**Comment by kikisdeliveryservice** [comment]
this is green ping @runcom

**Comment by runcom** [comment]
I fear this goes under:

>  Tech debt, refactors or code cleanups should not be
merged at this point. 

See eparis email about feature freeze, This will surely help us on moving towards ignv2s3 but since it's not scheduled for 4.1/4.2, this should go in in 4.2 like others e.g. #600.

**Comment by runcom** [comment]
needs another rebase

**Comment by kikisdeliveryservice** [comment]
want to look over the rebased changes a bit.

**Comment by kikisdeliveryservice** [comment]
ci woes...

/retest

**Comment by kikisdeliveryservice** [comment]
ok retesting again...

/retest

**Comment by kikisdeliveryservice** [comment]
@runcom finally green and fully rebased!

**Pull request merged**